web/api/v1/api_test.go

```diff
@@ -646,6 +649,90 @@ func TestReadEndpoint(t *testing.T) {
 	}
 }
 
+func TestSeriesEndpoint(t *testing.T) {
+	suite, err := promql.NewTest(t, `
+		load 1m
+			test_metric1{foo="bar"} 1+0x100
+			test_metric2{foo="bar"} 1+0x100
+	`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer suite.Close()
+
+	if err := suite.Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	readURL, err := url.Parse("http://localhost:8086/api/v1/prom/read?db=prometheus")
+	if err != nil {
+		t.Fatal(err)
+	}
+	logLevel := promlog.AllowedLevel{}
+	logLevel.Set("debug")
+	localStorage := suite.Storage()
+	l := log.With(promlog.New(logLevel), "component", "remote")
+	remoteStorage := remote.NewStorage(l, localStorage.StartTime, time.Minute)
+	conf := config.Config{
+		RemoteReadConfigs: []*config.RemoteReadConfig{
+			&config.RemoteReadConfig{
+				URL:           &config_util.URL{readURL},
+				RemoteTimeout: model.Duration(time.Second / 8),
+				ReadRecent:    false,
+			},
+		},
+	}
+	if err := remoteStorage.ApplyConfig(&conf); err != nil {
+		t.Fatal(err)
+	}
+
+	queryable := storage.NewFanout(promlog.New(logLevel), localStorage, remoteStorage)
+	api := &API{
+		Queryable:   queryable,
+		QueryEngine: suite.QueryEngine(),
+		config: func() config.Config {
+			return conf
+		},
+	}
+
+	end := time.Now().Unix()
+	start := end - 24*60*60
+	q := url.Values{
+		"match[]": []string{`test_metric1`},
+		"start":   []string{fmt.Sprintf("%d", start)},
+		"end":     []string{fmt.Sprintf("%d", end)},
+	}
+	seriesURL, err := url.Parse("http://example.com:9090/api/v1/series")
+	if err != nil {
+		t.Fatal(err)
+	}
+	seriesURL.RawQuery = q.Encode()
+	req, err := http.NewRequest("GET", seriesURL.String(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, apiErr, _ := api.series(req)
+	if apiErr != nil {
+		if apiErr.typ == errorExec {
```

Comment:
I realize you had a request on another PR to test the series/ endpoint, but having to whitelist various connection errors this way seems fragile.... I'd say that just testing the changed function (`ToQuery`) in `codec_test.go` would be ok.

Reply:
Thanks. I will cancel this PR.